app/models/role.rb

```diff
@@ -325,7 +325,7 @@ def filter_for_permissions_remove(resource_type, current_filters)
   end
 
   def find_current_filter(current_filters, filter_record)
-    current_filters.detect { |fil| fil.id == filter_record.id }
+    current_filters.reload.detect { |fil| fil.id == filter_record.id }
```

Comment:
This change makes me wonder about why this method was needed in the first place? The comment on line 315 seems to indicate that we are trying to use the filter that is in memory instead of the one in the database, but I don't understand why? I'm worried that we're missing some case here, since this changes makes us use the filters that are saved in the database instead of those that are in memory (and makes this whole method redundant).
I know most of this code is over 3 years old but perhaps @xprazak2 or @ares remember?

Reply:
I do not recall exactly - I just remember using the newly fetched record was causing problems with associations. On the other hand, the tests are passing and we have updated Rails since this was introduced, so maybe this works now even when reloading filters...